src/Symfony/Component/HttpKernel/Controller/ActionReference.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller;
+
+use Symfony\Component\HttpKernel\Bundle\BundleInterface;
+
+/**
+ * Class for holding bundle + controller name + action name.
+ *
+ * @author Pavel Batanov <pavel@batanov.me>
+ */
+class ActionReference
+{
+    /** @var BundleInterface */
+    public $bundle;
+    /** @var string */
+    public $controller;
+    /** @var string */
+    public $action;
```

Comment:
Should be private, otherwise they can be modified afterwards. I think this class should probably be final as well.

Reply:
Just followed https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/Controller/ControllerReference.php here. Can make it final VO to be more protected from modifications